### PR TITLE
Add support for logging extra columns in reward functions and update related tests

### PR DESCRIPTION
**Pull request by qgallouedec**
This PR uses the `log_extra` kwargs (added in #5233, cc @manueldeprada) in TRL's built-in reward functions, and add the console table to display extra columns.

## Changes

- `accuracy_reward`: logs `solution`, `gold_parsed`, and `answer_parsed` via `log_extra`
- `reasoning_accuracy_reward`: same extra columns, plus `"[incomplete reasoning]"` sentinel for completions missing the reasoning delimiter.

All new parameters default to `None` so the functions can still be called directly outside a trainer (e.g., in tests).

Screenshot 2026-03-18 at 6 43 23 PM

```python
from datasets import load_dataset
from trl import GRPOConfig, GRPOTrainer
from trl.rewards import accuracy_reward

dataset = load_dataset("trl-lib/DeepMath-103K", split="train").select(range(64))

trainer = GRPOTrainer(
    model="Qwen/Qwen2-0.5B-Instruct",
    reward_funcs=accuracy_reward,
    args=GRPOConfig(
        num_generations=4,
        max_completion_length=512,
        log_completions=True,
        logging_steps=1,
        max_steps=10,
    ),
    train_dataset=dataset,
)
trainer.train()
```

---

> [!NOTE]
> **Low Risk**
> Low risk: changes are limited to logging/diagnostic paths and optional parameters, with minimal impact on training logic aside from additional metadata collection.
> 
> **Overview**
> Adds support for displaying arbitrary *extra* per-completion metadata in the rich console completions table via a new `extra` argument to `print_prompt_completions_sample`, including subsampling behavior.
> 
> Extends `accuracy_reward` and `reasoning_accuracy_reward` to accept an optional `log_extra` callback and emit `solution`, `gold_parsed`, and `answer_parsed` (with sentinel values for unparseable/skipped/incomplete reasoning cases) for trainer logging.
> 
> Updates `GRPOTrainer` and `RLOOTrainer` to pass their collected `extra` logs into the console printer, and adds/updates tests to validate the new extra-column table output.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 69daa8008db4be552b139e382f3559b00a17e7be. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/rewards/accuracy_rewards.py [hunk 1]
```diff
@@ -86,18 +97,28 @@ def accuracy_reward(completions: list[list[dict[str, str]]], solution: list[str]
...
             # If the gold solution cannot be parsed, we assign `None` to skip this example
             reward = None
+            gold_parsed_strs.append("[unparseable]")
+            answer_parsed_strs.append("[skipped]")
         rewards.append(reward)
 
+    if log_extra is not None:
+        log_extra("solution", list(solution))
+        log_extra("gold_parsed", gold_parsed_strs)
+        log_extra("answer_parsed", answer_parsed_strs)
```
[the remark this answers is not in the document]
Comment (qgallouedec) [reply trl/rewards/accuracy_rewards.py:112]: no a fan of the suggestion. It's up to the user to ensure that the metrics name are unique.

**Comment by qgallouedec** [comment]
This PR is a minor thing and didn't get reviewed after 2 weeks, I'll go ahead and merge, feel free to suggest modification in subsequent PR is needed
